packages/fulfillment/src/services/fulfillment-module-service.ts

```diff
@@ -328,6 +350,59 @@ export default class FulfillmentModuleService<
     )
   }
 
+  async createShippingOptionRules(
+    data: FulfillmentTypes.CreateShippingOptionRuleDTO[],
+    sharedContext?: Context
+  ): Promise<FulfillmentTypes.ShippingOptionRuleDTO[]>
+  async createShippingOptionRules(
+    data: FulfillmentTypes.CreateShippingOptionRuleDTO,
+    sharedContext?: Context
+  ): Promise<FulfillmentTypes.ShippingOptionRuleDTO>
+
+  @InjectManager("baseRepository_")
+  async createShippingOptionRules(
+    data:
+      | FulfillmentTypes.CreateShippingOptionRuleDTO[]
+      | FulfillmentTypes.CreateShippingOptionRuleDTO,
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<
+    | FulfillmentTypes.ShippingOptionRuleDTO
+    | FulfillmentTypes.ShippingOptionRuleDTO[]
+  > {
+    const createdShippingOptionRules = await this.createShippingOptionRules_(
+      data,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<
+      | FulfillmentTypes.ShippingOptionRuleDTO
+      | FulfillmentTypes.ShippingOptionRuleDTO[]
+    >(createdShippingOptionRules, {
+      populate: true,
+    })
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  async createShippingOptionRules_(
+    data:
+      | FulfillmentTypes.CreateShippingOptionRuleDTO[]
+      | FulfillmentTypes.CreateShippingOptionRuleDTO,
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<TShippingOptionRuleEntity | TShippingOptionRuleEntity[]> {
+    const data_ = Array.isArray(data) ? data : [data]
+
+    if (!data_.length) {
+      return []
+    }
+
+    const createdShippingOptionRules =
```

Comment:
Any validation of the data we need to do? Eg. make sure the rule has at least one condition, etc?

Reply:
my bad, I ve missed to apply the validation here, only the db will reject for non nullable columns. I ll add that in